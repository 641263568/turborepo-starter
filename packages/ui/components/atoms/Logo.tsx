import Link from "next/link";

export interface LogoImage {
  as: "image";
  alt: string;
  title: string;
  el: any;
}

export interface LogoText {
  as: "text";
  title: string;
}

export type LogoProps = LogoImage | LogoText;

export const Logo = (logo: LogoProps) => {
  if (logo.as === "text") {
    return (
      <div className="dark:text-white text-black">
        <Link href="/">{logo.title}</Link>
      </div>
    );
  }

  return (
    <Link href="/">
      <a className="dark:text-white text-black">
        {logo.el}
        <span className="sr-only">{logo.title}</span>
      </a>
    </Link>
  );
};

Logo.displayName = "Logo";
