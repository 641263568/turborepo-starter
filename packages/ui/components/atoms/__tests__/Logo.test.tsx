import { render, screen } from "@testing-library/react";
import { Logo } from "../Logo";
import { axe, toHaveNoViolations } from "jest-axe";

expect.extend(toHaveNoViolations);

jest.mock("next/link", () => {
  return ({ children }: any) => {
    return children;
  };
});

test("logo as text passes a11y test", async () => {
  const { container } = render(<Logo as="text" title="Emry" />);
  expect(await axe(container)).toHaveNoViolations();
});

it("logo as text renders unchanged", () => {
  const { container } = render(<Logo as="text" title="Emry" />);
  expect(container).toMatchSnapshot();
});

it("it renders the logo as a title", () => {
  const { getByText } = render(<Logo as="text" title="Emry" />);
  expect(getByText("Emry")).toBeTruthy();
});

it("logo as text has dark mode", () => {
  render(<Logo as="text" title="Emry" />);
  expect(screen.getByText("Emry")).toHaveClass("dark:text-white");
});
