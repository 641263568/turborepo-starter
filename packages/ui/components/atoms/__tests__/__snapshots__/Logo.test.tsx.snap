// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logo as text renders unchanged 1`] = `
<div>
  <div
    class="dark:text-white text-black"
  >
    Emry
  </div>
</div>
`;
